marketplace/clients/facebook/client.py

```diff
@@ -223,5 +223,24 @@ def get_wpp_commerce_settings(self, wa_phone_number_id):
     def get_template_analytics(self, waba_id, fields):
         url = self.BASE_URL + f"{waba_id}/template_analytics"
         headers = self._get_headers()
-        response = self.make_request(url, method="GET", headers=headers, params=fields)
-        return response.json()
+        combined_data = {}
+
+        while url:
+            response = self.make_request(
+                url, method="GET", headers=headers, params=fields
+            )
+            data = response.json()
+
+            for entry in data.get("data", []):
+                data_points = entry.get("data_points", [])
+
+                if "data" not in combined_data:
+                    combined_data["data"] = {"data_points": []}
+
+                combined_data["data"]["data_points"].extend(data_points)
+
+            paging = data.get("paging", {})
+            next_url = paging.get("next")
+            url = next_url if next_url else None
```

Comment:
This way you can reduce the size of the code and make it less complex.

```suggestion
            url = paging.get("next", None)
```